softioc/asyncio_dispatcher.py

```diff
@@ -47,6 +48,25 @@ def close(self):
             self.__atexit = None
 
         self.__shutdown()
+    
+    def wait_for_quit(self):
+
+        async_stop_event = asyncio.Event()
+        stop_event = threading.Event()
+
+        def signal_exit():
+            self.loop.call_soon_threadsafe(async_stop_event.set)
+        
+        async def async_signal_exit():
+            await async_stop_event.wait()
+            stop_event.set()
+
+        for sig in ('SIGINT', 'SIGTERM'):
+            signal.signal(getattr(signal, sig), lambda signum, frame: signal_exit())
```

Comment:
Also, why not just let `signal_exit` take `signum` and `frame` as arguments, then this loop is simply:
```python
for sig in (signal.SIGINT, signal.SIGTERM):
    signal.signal(sig, signal_exit)
```

Reply:
I've tested this proposal, but I get an error message when I close the program. 

```
python test.py 
INFO: PVXS QSRV2 is loaded, permitted, and ENABLED.
Starting iocInit
############################################################################
## EPICS 7.0.7.0
## Rev. 7.0.7.99.0.2
## Rev. Date 7.0.7.99.0.2
############################################################################
iocRun: All initialization complete
^CTraceback (most recent call last):
  File "/home/xavier/test.py", line 29, in <module>
    softioc.non_interactive_ioc()
  File "/home/xavier/test_softioc/lib64/python3.12/site-packages/softioc/softioc.py", line 365, in non_interactive_ioc
    device.dispatcher.wait_for_quit()
  File "/home/xavier/test_softioc/lib64/python3.12/site-packages/softioc/asyncio_dispatcher.py", line 72, in wait_for_quit
    stop_event.wait()
  File "/usr/lib64/python3.12/threading.py", line 655, in wait
    signaled = self._cond.wait(timeout)
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.12/threading.py", line 355, in wait
    waiter.acquire()
TypeError: AsyncioDispatcher.wait_for_quit.<locals>.signal_exit() takes 0 positional arguments but 2 were given
```